src/stenotype/backend/elements.py

```diff
@@ -0,0 +1,199 @@
+"""
+Steno Type Elements
+
+Representations of the fundamental :py:mod:`stenotype` building blocks.
+Each element holds the relevant information for both a :py:mod:`stenotype` and
+:py:mod:`typing` conversion; ``str(element)`` provides the canonical
+:py:mod:`stenotype` representation.
+
+Note that all elements are represented as values, not types:
+A :py:class:`Steno` encodes :py:mod:`stenotype` syntax/code,
+not any concrete types.
+For example, ``Identifier('typing', 'List')`` is not the type :py:class:`typing.List`
+but represents the name ``typing.List``.
+"""
+from typing import NamedTuple, Tuple as TupleT, Union as UnionT
+
+
+class Dots:
+    """A literal ``...`` placeholder"""
+
+    __slots__ = ()
+
+    def __eq__(self, other):
+        return True if type(other) is Dots else NotImplemented
+
+    def __str__(self):
+        return "..."
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}()"
+
+
+class Identifier(TupleT[str, ...]):
+    """A qualified name, such as ``typing.Tuple``"""
+
+    def __new__(cls, *elements: str):
+        assert all(type(el) is str and el.isidentifier() for el in elements)
```

Comment:
To get past bandit, this should work
```suggestion
        assert all(type(el) is str and el.isidentifier() for el in elements)  # nosec
```
Or rewriting it as a `TypeError` or `StenotypeException`, if that makes sense. I'd personally prefer any of those, but if you like to use asserts then I can push a quick config change as well.

Reply:
Since it is the only sanity check at the moment, I'm in favour of just removing it. It's what type checkers are there for, after all (oh, the irony...).